File: app/src/main/java/com/trinity/sample/MainActivity.java
package com.trinity.sample;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'native-lib' library on application startup.
    static {
        System.loadLibrary("trinity");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button videoRecordButton = findViewById(R.id.forward_video_recorder);
        videoRecordButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, VideoRecorderActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/trinity/sample/MainActivity.kt
package com.trinity.sample

import android.Manifest
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import com.github.florent37.runtimepermission.RuntimePermission
import com.tencent.mars.xlog.Xlog
import com.trinity.sample.entity.MediaItem

class MainActivity : AppCompatActivity() {

    companion object {
        init {
            System.loadLibrary("trinity")
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        requestPermission()

        findViewById<Button>(R.id.record)
            .setOnClickListener {
                val intent = Intent(this, RecordActivity::class.java)
                startActivity(intent)
            }

        findViewById<Button>(R.id.editor)
            .setOnClickListener {
                val intent = Intent(this, EditorActivity::class.java)
                val mediaItem = MediaItem("/sdcard/DCIM/Camera/no_audio.mp4", "", 540, 960)
                mediaItem.duration = 14000
                val mediaItems = mutableListOf<MediaItem>()
                mediaItems.add(mediaItem)
                intent.putExtra("medias", mediaItems.toTypedArray())
                startActivity(intent)
            }

        findViewById<Button>(R.id.music)
            .setOnClickListener {
                val intent = Intent(this, MusicActivity::class.java)
                startActivity(intent)
            }
    }

    private fun requestPermission() {
        RuntimePermission.askPermission(this, Manifest.permission.CAMERA,
            Manifest.permission.RECORD_AUDIO,
            Manifest.permission.WRITE_EXTERNAL_STORAGE,
            Manifest.permission.READ_EXTERNAL_STORAGE)
            .onAccepted {

            }
            .onDenied {

            }
    }

}
